kwave/utils/deprecation.py

```diff
@@ -0,0 +1,35 @@
+"""Utilities for handling deprecation warnings in k-Wave."""
+
+import functools
+import warnings
+from typing import Callable, TypeVar, Any
+
+F = TypeVar("F", bound=Callable[..., Any])
+
+
+def deprecated(message: str, target_version: str = "2.0.0") -> Callable[[F], F]:
+    """Decorator to mark functions as deprecated.
+
+    Args:
+        message: Message explaining what to use instead
+        target_version: Version in which the function will be removed
+
+    Example:
+        @deprecated("Use new_function() instead", "2.0.0")
+        def old_function():
+            pass
+    """
+
+    def decorator(func: F) -> F:
```

Comment:
Usually these type of calls happen once per application lifecycle. If we display warning every time the function is called, it could be annoying. Perhaps move the warning outside of the `wrapper` but keep it still inside of the `decorator`? Since each function gets decorated only once (not sure how this will if the file is imported multiple times), a single log should be printed.

Reply:
Thanks. I've updated the wrapper.

Reply:
Actually, will this be printing the deprecation warning right away? Even if the function is not used?

If yes, let's move it back and introduce a variable that checks if the warning was printed or not?

Reply:
I migrated to the deprecation package so this is standardized